apps/checkout/src/hooks/useErrors/useErrors.ts

```diff
@@ -0,0 +1,51 @@
+import { useGetParsedApiErrors } from "./useGetParsedApiErrors";
+import { useState } from "react";
+import { ApiErrors, Errors } from "./types";
+
+export interface UseErrors<TFormData> {
+  errors: Errors<TFormData>;
+  setErrors: (errors: Errors<TFormData>) => void;
+  setApiErrors: (apiErrors: ApiErrors<TFormData>) => void;
+  clearErrors: () => void;
+  hasErrors: boolean;
+}
+
+export const useErrors = <TFormData>(): UseErrors<TFormData> => {
+  const [errors, setErrors] = useState<Errors<TFormData>>({});
+  const getParsedApiErrors = useGetParsedApiErrors();
+
+  const getParsedErrors = <TFormData>(apiErrors: ApiErrors<TFormData>) => {
+    if (!apiErrors) {
+      return {} as Errors<TFormData>;
+    }
+
+    // @ts-ignore to be fixed before merge
+    return getParsedApiErrors(apiErrors).reduce(
+      (result, { field, ...rest }) => {
+        return {
+          ...result,
+          [field]: {
+            field,
+            ...rest,
+          },
+        };
+      },
+      {} as Errors<TFormData>
+    );
+  };
+
+  const setApiErrors = (apiErrors: ApiErrors<TFormData>) =>
+    setErrors(getParsedErrors(apiErrors));
```

Comment:
I don't know if it is a case in checkout but what would happen if there would be 2 or more consecutive queries that result in an error? Wouldn't the newest one overwrite the old one?